src/argparse.c

```diff
@@ -207,12 +342,35 @@ void neu_cli_args_init(neu_cli_args_t *args, int argc, char *argv[])
 
     memset(args, 0, sizeof(*args));
 
-    // load env
-    load_env(args, &log_level, &config_dir, &plugin_dir);
-
     int c            = 0;
     int option_index = 0;
 
+cfile:
```

Comment:
also about readability, the `goto` statement combined with the `switch` statement makes the code difficult to comprehend, maybe it is better to parse the `config_file` option in a separate function, and then these `if` checks could be resided in that function ?